sql_script/MS3_insertions.sql

```diff
@@ -1,41 +1,42 @@
-/*INSERTIONS___
-#Add new inventory types to the inventory_type table*/
-INSERT
-INTO inventory_type(type_name, height, width, length)
-VALUES (new_type_name, new_height, new_width, new_length);#impact on dimensions table?
-
-/*Add a new recipient to the recipient table*/
-INSERT
-INTO recipient(id, street_address, postal_code)
+/*Add new inventory types to the inventory_type table 
+-> NOTE: only allows insertion when dimensions already exist in dimensions table*/
+INSERT INTO inventory_type(type_name, height, width, length)
+VALUES (new_type_name, new_height,new_width,new_length);
+
+/*Add a new recipient to the recipient table 
+-> NOTE: only allows insertion if postal code already exists in region*/
+INSERT INTO recipient(id, street_address, postal_code)
 VALUES (new_id, new_street_address, new_postal_code);
 
-/*Add a new producer to the producer table*/
+/*Add a new producer to the producer table
+-> NOTE: only works when new_postal_code already exists in region table*/
 INSERT
 INTO producer (id, street_address, postal_code)
 VALUES (new_id, new_street_address, new_postal_code);
 
-/*!!!!!!!!!!!!!!!Come up with values for a new container type, calculate its volume, insert that value into the dimensions table, then insert the container into container_type (has a FK to dimensions).
-*/
-INSERT
-INTO storage_container (id, warehouse_id, occupied_volume, height, width, length)
-VALUES (new_id, new_warehouse_id, 0, new_height, new_width, new_length);
+/*Come up with values for a new container type, calculate its volume, insert that value into the dimensions table, then insert the container into container_type (has a FK to dimensions).
+-> NOTE: only works if height, width, length tuple already exists in dimension table*/
```

Comment:
Implement check in back-end and add tuple to dimensions if it doesn't already exist.

Reply:
You'll want to run this query beforehand:

```sql
INSERT INTO dimensions
VALUES (new_height, new_width, new_length, new_height * new_width * new_height)
ON CONFLICT (height, width, length) DO NOTHING;
```

The value will be substituted for $1, $2, $3, $4 in the code.